### ISSUE TITLE
Draw a box to zoom in

### ISSUE THREAD

**Issue by sonnyk22** [maurizzzio/function-plot#17]
Is there an option to allow us to draw a box anywhere on the graph which will zoom into only the box area?  This is done using the mouse, by holding the mouse (mouse-down) to draw a box, and then will zoom in when the mouse (mouse-up) button has been released.  Or, by clicking on a button somewhere on the body tag to tell the D3 chart to allow us to draw a box.

Please see the example below.

![draw box to zoom](https://cloud.githubusercontent.com/assets/6744710/8907775/2f9cc3f0-3446-11e5-8de3-0f5a9cfa8951.png)

**Comment by maurizzzio** [comment]
Not implemented yet

**Comment by sonnyk22** [comment]
I came across below links that may help:

http://stackoverflow.com/questions/18273884/live-drawing-of-a-line-in-d3-js
http://bl.ocks.org/lgersman/5311083

**Comment by maurizzzio** [comment]
thanks! well I saw that this is actually implemented in plot.ly however I'm working on something else right now, perhaps I can do more research in the weekend

**Comment by sonnyk22** [comment]
that would be awesome!  thank you for all your help;  this is a great library you have put together.

**Comment by maurizzzio** [comment]
Implemented on v1.8.0, check out the front page for an example of a plugin

e.g.

```javascript
functionPlot({
  target: '#plugin-zoom-box',
  data: [{
    fn: 'x^2'
  }],
  plugins: [
    functionPlot.plugins.zoomBox({
      key: '<shift>'
    })
  ]
})
```

**Comment by sonnyk22** [comment]
Awesome!  It works great!  Thank you
